src/main.cpp

```diff
@@ -2906,33 +2908,63 @@ TbBool setup_move_out_of_cave_in(struct Thing *thing)
     {
         struct CreatureControl* cctrl;
         cctrl = creature_control_get_from_thing(thing);
-        if ( setup_person_move_to_coord(thing, &cctrl->flee_pos, 0) )
+        long dist = LbDiagonalLength(abs(thing->mappos.x.val - cctrl->flee_pos.x.val), abs(thing->mappos.y.val - cctrl->flee_pos.y.val));
+        // If you're too close to the flee position, no point in going there to escape cave in damage.
+        if (dist <= 200)
         {
-            return true;
+            // Heroes that are near to a hero gate, should escape through it if they can.
+            if (is_hero_thing(thing))
+            {
+                if (good_leave_through_exit_door(thing))
```

Comment:
Isnt it too cheesy?

Reply:
Old behavior: They will flee to the hero gate, then stand still at the hero gate forever taking cave in damage. This allows you to round up heroes and kill them all with cave-in.
New behavior: When you chase heroes towards the hero-gate, and cast CTA again, they will stay out for a few seconds to avoid the cave-in damage.

Do you see something wrong with the new behavior?

Reply:
Hero will never actually leave using this door.

Reply:
Which is the expected behavior.